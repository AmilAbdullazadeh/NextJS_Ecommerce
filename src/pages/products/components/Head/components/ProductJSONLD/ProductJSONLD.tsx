import React from 'react';

import { ApiProduct } from '@api';
import { absoluteURL, productDetailsPage, productImageUrl } from '@core';

export interface ProductJSONLDprops {
    product: ApiProduct;
}

const ProductJSONLD: React.FC<ProductJSONLDprops> = ({
    product,
}: ProductJSONLDprops) => {
    const JSON_LD = {
        '@context': 'http://schema.org',
        '@type': 'Product',
        aggregateRating: product.ratingValue ?? 0,
        brand: product.brand,
        category: 'Mobile Phones',
        image: absoluteURL(productImageUrl(product.image)),
        model: product.name,
        name: product.name,
        url: absoluteURL(productDetailsPage(product.id)),
    };

    return (
        <script
            type="application/ld+json"
            dangerouslySetInnerHTML={{ __html: JSON.stringify(JSON_LD) }}
        ></script>
    );
};

export default ProductJSONLD;
