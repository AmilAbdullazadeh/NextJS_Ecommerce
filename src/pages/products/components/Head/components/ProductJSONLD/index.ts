export { default as ProductJSONLD } from './ProductJSONLD';
