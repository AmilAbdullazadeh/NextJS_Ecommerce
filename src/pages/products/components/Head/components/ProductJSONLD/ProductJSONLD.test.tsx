import React from 'react';

import { render } from '@testUtils';

import ProductJSONLD, { ProductJSONLDprops } from './ProductJSONLD';

describe('ProductJSONLD', () => {
    const defaultsProps: ProductJSONLDprops = {
        product: {
            id: 'mock id',
            sku: 'mock sku',
            brand: 'mock brand',
            price: 5,
            availability: 5,
            description: 'mock description',
            name: 'mock name',
            image: '',
        },
    };

    it('renders correctly', () => {
        const {
            container: { firstChild },
        } = render(<ProductJSONLD {...defaultsProps} />);

        expect(firstChild).toMatchSnapshot();
    });
});
