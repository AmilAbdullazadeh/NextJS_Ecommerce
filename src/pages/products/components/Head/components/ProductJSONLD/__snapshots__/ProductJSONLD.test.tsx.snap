// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductJSONLD renders correctly 1`] = `
<script
  type="application/ld+json"
>
  {"@context":"http://schema.org","@type":"Product","aggregateRating":0,"brand":"mock brand","category":"Mobile Phones","image":"http://localhost:3000/images/","model":"mock name","name":"mock name","url":"http://localhost:3000/products/mock%20id"}
</script>
`;
